salt/modules/file.py

```diff
@@ -6004,39 +6004,114 @@ def manage_file(
+        swift://container/object_path URLs, see swift.get documentation.
+        For files hosted on the salt file server, if the file is located on
+        the master in the directory named spam, and is called eggs, the source
+        string is salt://spam/eggs. If source is left blank or None
+        (use ~ in YAML), the file will be created as an empty file and
+        the content will not be managed. This is also the case when a file
+        already exists and the source is undefined; the contents of the file
+        will not be changed or managed. If source is left blank or None, please
+        also set replaced to False to make your intention explicit.
+
+
+        If the file is hosted on a HTTP or FTP server then the source_hash
+        argument is also required.
 
     source_sum
         sum hash for source
 
     user
-        user owner
+        The user to own the file, this defaults to the user salt is running as
+        on the minion
 
     group
-        group owner
+        The group ownership set for the file, this defaults to the group salt
+        is running as on the minion. On Windows, this is ignored
 
-    backup
-        backup_mode
+    mode
+        The permissions to set on this file, e.g. ``644``, ``0775``, or
+        ``4664``.
+
+        The default mode for new files and directories corresponds to the
+        umask of the salt process. The mode of existing files and directories
+        will only be changed if ``mode`` is specified.
+
+        .. note::
+            This option is **not** supported on Windows.
 
     attrs
-        attributes to be set on file: '' means remove all of them
+        The attributes to have on this file, e.g. ``a``, ``i``. The attributes
+        can be any or a combination of the following characters:
+        ``aAcCdDeijPsStTu``.
+
+        .. note::
+            This option is **not** supported on Windows.
 
         .. versionadded:: 2018.3.0
 
+    saltenv
+        Specify the environment from which to retrieve the file indicated
+        by the ``source`` parameter. If not provided, this defaults to the
+        environment from which the state is being executed.
+
+        .. note::
+            Ignored when the source file is from a non-``salt://`` source..
+
+    backup
+        Overrides the default backup mode for this specific file. See
+        :ref:`backup_mode documentation <file-state-backups>` for more details.
+
     makedirs
-        make directories if they do not exist
+        If set to ``True``, then the parent directories will be created to
+        facilitate the creation of the named file. If ``False``, and the parent
+        directory of the destination file doesn't exist, the state will fail.
 
     template
-        format of templating
+        If this setting is applied, the named templating engine will be used to
+        render the downloaded file. The following templates are supported:
+
+        - :mod:`cheetah<salt.renderers.cheetah>`
```

Comment:
While these are the currently supported it should be noted that custom renders that output string are also supported.